### PR TITLE
Backend dev registration bug fixes

### PR DESCRIPTION
**Pull request by alperendivriklioglu**
This push contains the necessary bug fixes for the issues #173 , #195 and #176 .

### CODE REVIEW

File: Platon/backend/app/auth_system/views.py [hunk 1]
```diff
@@ -191,16 +190,18 @@ def get(requester_id, self):
                     if Follow.query.filter_by(follower_id=requester_id, following_id=form.user_id.data).first() is not None:
```
Comment (alperendivriklioglu) [inline Platon/backend/app/auth_system/views.py:191]: @oykuyilmaz1 @darktheorys @ahmetdadak 

After this pull request is merged and deployed to our server, you will be able to get whether the requester user follows the requested user or not when you send a "Get User" request.

For example:
Requester user: User 1
Requested user: User 2

If User 1 already follows User 2, "following_status" field will have the value "1".
If User 1 has sent a follow request to User 2; but User 2 has not yet responded to that request, "following_status" field will have the value "0".
If User 1 is not currently following User 2 and does not have a pending follow request, "following_status" field will have the value "-1".

Comment (alperendivriklioglu) [inline Platon/backend/app/auth_system/views.py:191]: @oykuyilmaz1 @darktheorys @ahmetdadak 

After this pull request is merged and deployed to our server, you will be able to get whether the requester user follows the requested user or not when you send a "Get User" request.

For example:
Requester user: User 1
Requested user: User 2

If User 1 already follows User 2, "following_status" field will have the value "1".
If User 1 has sent a follow request to User 2; but User 2 has not yet responded to that request, "following_status" field will have the value "0".
If User 1 is not currently following User 2 and does not have a pending follow request, "following_status" field will have the value "-1".

File: Platon/backend/app/auth_system/views.py [hunk 2]
```diff
@@ -191,16 +190,18 @@ def get(requester_id, self):
...
                         following_status = 1 # Represents that the requester user follows the requested user.
                     elif FollowRequests.query.filter_by(follower_id=requester_id, following_id=form.user_id.data).first() is not None:
-                        following_status = 0 # Represents that the requester user has already sent a follow request to the requested user.
+                        following_status = 0 # Represents that the requester user has already sent a follow request to the requested user and it is pending.
                     else:
                         following_status = -1 # Represents that the requester user does not follow the requested user and has not yet sent a request to follow them.
```
Comment (alperendivriklioglu) [inline Platon/backend/app/auth_system/views.py:197]: @oykuyilmaz1 @darktheorys @ahmetdadak 

After this pull request is merged and deployed to our server, you will be able to get whether the requester user follows the requested user or not when you send a "Get User" request.

For example:
Requester user: User 1
Requested user: User 2

If User 1 already follows User 2, "following_status" field will have the value "1".
If User 1 has sent a follow request to User 2; but User 2 has not yet responded to that request, "following_status" field will have the value "0".
If User 1 is not currently following User 2 and does not have a pending follow request, "following_status" field will have the value "-1".

**Review by cakmadam98 — APPROVED** [review]
Looks cool

**Review by hilal-demir — APPROVED** [review]

File: Platon/backend/app/auth_system/views.py [hunk 3]
```diff
@@ -173,22 +180,32 @@ def get(self):
...
+
                     account_information = { 
                                         "id": existing_user.id,
                                         "name": existing_user.name,
                                         "surname": existing_user.surname,
+                                        "e_mail": existing_user.e_mail,
+                                        "is_private": existing_user.is_private,
+                                        "following_status": following_status,
                                         "rate": existing_user.rate,
                                         "profile_photo": existing_user.profile_photo,
```
Comment (HUmutOzdemir) [inline Platon/backend/app/auth_system/views.py:208]: Why there is no field that represents the institution of the user. As far as I know, in the last Backend meeting we decided to add this part to the user endpoints. Please make the adaquate changes on modes and current code.

**Review by HUmutOzdemir — CHANGES_REQUESTED** [review]
Is there an edition for uploading profile photos? I can not see them in this PR. When do you plan to add such a functionality?

File: Platon/backend/app/auth_system/views.py (hunk 3 above)
Comment (HUmutOzdemir) [reply Platon/backend/app/auth_system/views.py:208]: As I can see, @hilal-demir added this feature in her PR. So it is resolved.

**Pull request merged**